activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -118,11 +118,17 @@ def index_by
   #   %i( title body ).index_with { |attr_name| post.public_send(attr_name) }
   #   # => { title: "hey there", body: "what's up?" }
   #
-  # If an argument is passed instead of a block, it will be used as the value
+  # If an argument is passed instead of a block, the same instance will be used as the value
   # for all elements:
   #
   #   %i( created_at updated_at ).index_with(Time.now)
   #   # => { created_at: 2020-03-09 22:31:47, updated_at: 2020-03-09 22:31:47 }
+  #   %i( a_key a_second_key ).index_with({})
+  #   # => { a_key: {}, a_second_key: {} }
+  #
+  # In the latter example, `index_with` is using the same object instance as the value of both keys.
+  # A change to one of the hashes is a change on both. If this is not desired, use the block
+  # variant instead.
```

Comment:
I think it would be clearer to move the new example to a separate block.
Maybe also use a less abstract example.
What do you think about something like:
```suggestion
  #
  # The argument passed to `index_with` is using the same object instance for all values.
  # A change to the variable is a change on all values:
  #
  #   pets = %i( cats dogs ).index_with([])
  #   pets[:cats] << 'Maceo'
  #   pets
  #   # => {:cats=>["Maceo"], :dogs=>["Maceo"]}
  #  
  # If this is not desired, use the block variant instead.
```

Reply:
That looks great!

Reply:
Awesome, I'll apply the changes.